### PR TITLE
fix(workspace): correct title generartion for notes with sub-hierarchy starting with .md

### PR DESCRIPTION
**Pull request by Harshita-mindfire**
### Basics
- [x] code should follow [[Code Conventions|dendron://dendron.docs/dev.process.code]]
- [x] circular dependency check: make sure your code is not introducing new circular dependencies in plugin-core.  See [[Avoiding Circular Dependencies|dendron://dendron.docs/dev.process.code.best-practices#avoiding-circular-dependencies]].
- [ ] sticking to existing conventions instead of creating new ones
    - eg: [if configuration for utilities are already in one module or package, add future utilities there as well](https://github.com/dendronhq/dendron/pull/1960#discussion_r786228021)

### Extended
- General
    - [x] check whether code be simplified
    - [ ] check if similar function already exist in the codebase. if so, can it be re-used?
    - [ ] check if this change adversely impact performance
- Operations
    - [ ] when shipping this change, will it just work or will it introduce additional operational overhead due to complicated interface or known bugs?
- Architecture
    - [ ] check if code is introducing changes on a foundational class or interface. if so, call for design review if needed
        - eg: [making changes to DNode](https://github.com/dendronhq/dendron/pull/2158#pullrequestreview-854689586)

### CODE REVIEW

**Comment by Harshita-mindfire** [comment]
Related issue: https://github.com/dendronhq/dendron/issues/1972 and https://github.com/dendronhq/dendron/issues/856

madge report: now new circular dependencies(64)

File: packages/common-all/src/dnode.ts [hunk 1]
```diff
@@ -94,7 +94,8 @@ export class DNodeUtils {
   }
 
   static basename(nodePath: string, rmExtension?: boolean) {
-    if (rmExtension) {
+    //skip for nodePath that has a sub-hierarchy starting with .md eg: learn.mdone.test, learn.md-one.new
+    if (rmExtension && nodePath.endsWith(".md")) {
```
Comment (SeriousBug) [inline packages/common-all/src/dnode.ts:98]: minor nitpick: could simplify this the rest of this part, if path ends with ".md" then `nodePath = nodePath.slice(undefined, -3)`

**Review by SeriousBug — APPROVED** [review]
Looks good, left one nitpick on simplifying the code.

File: packages/common-all/src/dnode.ts (hunk 1 above)
Comment (Harshita-mindfire) [reply packages/common-all/src/dnode.ts:98]: Updated. Thanks!

**Review by kevinslin — APPROVED** [review]

**Pull request merged**